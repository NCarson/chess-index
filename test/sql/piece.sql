
select 'p'::cpiece::piece::text = 'P';

select pretty('p'::piece);
select pretty('n'::piece);
select pretty('b'::piece);
select pretty('r'::piece);
select pretty('q'::piece);
select pretty('k'::piece);

select expected_or_fail_bool('p'::piece = 'P'::piece, true);
select expected_or_fail_bool('p'::piece != 'P'::piece, false);
select expected_or_fail_bool('p'::piece != 'K'::piece, true);
select expected_or_fail_bool('p'::piece = 'K'::piece, false);

select expected_or_fail_bool(value('P'::piece) =  1, true);
select expected_or_fail_bool(value('N'::piece) =  3, true);
select expected_or_fail_bool(value('B'::piece) =  3, true);
select expected_or_fail_bool(value('R'::piece) =  5, true);
select expected_or_fail_bool(value('Q'::piece) =  9, true);
select expected_or_fail_bool(value('K'::piece) =  0, true);
