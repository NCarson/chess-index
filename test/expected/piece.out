select 'p'::cpiece::piece::text = 'P';
 ?column? 
----------
 t
(1 row)

select pretty('p'::piece);
 pretty 
--------
 ♙
(1 row)

select pretty('n'::piece);
 pretty 
--------
 ♘
(1 row)

select pretty('b'::piece);
 pretty 
--------
 ♗
(1 row)

select pretty('r'::piece);
 pretty 
--------
 ♖
(1 row)

select pretty('q'::piece);
 pretty 
--------
 ♕
(1 row)

select pretty('k'::piece);
 pretty 
--------
 ♔
(1 row)

select expected_or_fail_bool('p'::piece = 'P'::piece, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('p'::piece != 'P'::piece, false);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('p'::piece != 'K'::piece, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('p'::piece = 'K'::piece, false);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(value('P'::piece) =  1, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(value('N'::piece) =  3, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(value('B'::piece) =  3, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(value('R'::piece) =  5, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(value('Q'::piece) =  9, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(value('K'::piece) =  0, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

